Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/aspect/EventAspectAdvice.java

```diff
@@ -47,123 +48,124 @@
+     * invocation. This will allow the service to both log the exception raised, log the failed processing event, suppress
+     * logging the corresponding END_* event, and satisfy the requirements of the given service spec by returning a
+     * RegistryErrorList containing the errors that occurred.
+     *
+     * If the exception is any other type other than ErrorEventException, the event will be logged as a failure, with
+     * no corresponding END_* event and the exception will be re-thrown.
+     *
+     * @param joinPoint
+     * @param delegate
+     * @param beforeBuilder
+     * @param afterBuilder
+     * @param serviceType
+     * @param version
+     * @return
+     * @throws Throwable
      */
-    @AfterThrowing(pointcut = "@annotation(gov.hhs.fha.nhinc.aspect.InboundMessageEvent) ||"
-        + "@annotation(gov.hhs.fha.nhinc.aspect.InboundProcessingEvent) ||"
-        + "@annotation(gov.hhs.fha.nhinc.aspect.AdapterDelegationEvent) || "
-        + "@annotation(gov.hhs.fha.nhinc.aspect.OutboundMessageEvent) || "
-        + "@annotation(gov.hhs.fha.nhinc.aspect.OutboundProcessingEvent) || "
-        + "@annotation(gov.hhs.fha.nhinc.aspect.NwhinInvocationEvent)", throwing = "e")
-    public void failEvent(JoinPoint joinPoint, Throwable e) {
-        if (eventRecorder != null && eventRecorder.isRecordEventEnabled()) {
-            MethodSignature sig = (MethodSignature) joinPoint.getSignature();
-            Method method = sig.getMethod();
-
-            // Check if the method is annotated with @LogFailures. We dont want to log the event twice.
-            if (method.getAnnotation(LogFailures.class) != null) {
-                Logger log = LoggerFactory.getLogger(sig.getDeclaringType()); //NOSONAR
-                log.warn("Method {} has multiple logging aspects. Will only process LogFailures annotation... ", method.getName());
-            } else {
-                logFailure(joinPoint, e);
-            }
+    public Object logEvents(ProceedingJoinPoint joinPoint, EventAdviceDelegate delegate, Class<? extends EventDescriptionBuilder> beforeBuilder,
+        Class<? extends EventDescriptionBuilder> afterBuilder, String serviceType, String version) throws Throwable {
+        delegate.begin(joinPoint.getArgs(), serviceType, version, beforeBuilder);
+
+        Object value = null;
+        boolean hasFailure = false;
+        try {
+            value = joinPoint.proceed();
+        } catch (ErrorEventException e) {
+            // Log the failure, and handle the exception. Service should still return a value in this case
+            logFailure(joinPoint, e, serviceType, version);
+            value =  e.getReturnOverride();
+            hasFailure = true;
+        } catch (Throwable e) {
+            // We don't want to swallow the exception if its not an ErrorEventException.
+            // We also don't want the delegate to log the END_* event.
+            logFailure(joinPoint, e, serviceType, version);
+            throw e;
```

Comment:
if this throw, we never hit the logic below

Reply:
That is the point. If we swallow the exception, then there will be an END_* event logged into the database. Per the Event Logging Approach wiki page, we want either a BEGIN_* and a corresponding END_*, or a BEGIN_* and a MESSAGE_PROCESSING_FAILED event. If we swallow the exception and ignore it, we will have all 3: BEGIN_*, MESSAGE_PROCESSING_FAILED, and END_*